common/src/main/java/com/wynntils/features/utilities/MythicSellWarningFeature.java

```diff
@@ -0,0 +1,348 @@
+import com.wynntils.utils.mc.LoreUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.FontRenderer;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.type.HorizontalAlignment;
+import com.wynntils.utils.render.type.TextShadow;
+import com.wynntils.utils.render.type.VerticalAlignment;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.gui.screens.inventory.ContainerScreen;
+import net.minecraft.client.resources.language.I18n;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.ItemStack;
+import net.neoforged.bus.api.EventPriority;
+import net.neoforged.bus.api.SubscribeEvent;
+
+@ConfigCategory(Category.UTILITIES)
+public class MythicSellWarningFeature extends Feature {
+    @Persisted
+    public final Config<Boolean> tomesWarning = new Config<>(false);
+
+    @Persisted
+    public final Config<Float> tradeMarketPriceThreshold = new Config<>(0.9f);
+
+    private static final ResourceLocation CIRCLE_TEXTURE =
+            ResourceLocation.withDefaultNamespace("textures/wynn/gui/tutorial.png");
+
+    private static final String BLACKSMITH_TITLE = "\uDAFF\uDFF8\uE016";
+    private static final int BLACKSMITH_CONFIRM_BUTTON_SLOT = 17;
+
+    private static final String TM_SELL_TITLE = "\uDAFF\uDFE8\uE014\uDAFF\uDF951";
+    private static final int TM_ITEM_SLOT = 22;
+    private static final int TM_SELL_PRICE_SLOT = 28;
+    private static final Pattern TM_SELL_PRICE_PATTERN = Pattern.compile("§a- §7Total:§f (\\d{1,3}(?:,\\d{3})*)");
+    private static final int TM_PRICE_CHECK_SLOT = 51;
+    private static final Pattern TM_PRICE_CHECK_PATTERN =
+            Pattern.compile("§7Cheapest Sell Offer: §f(\\d{1,3}(?:,\\d{3})*)");
+    private static final int TM_CONFIRM_BUTTON_SLOT = 34;
+
+    private HintTextWidget ctrlHintTextWidget;
+    private final List<HintTextWidget> tmHintTextWidgets = new ArrayList<>();
+    private int emphasizeAnimationFrame = 0; // 0-indexed 4 frames of animation
+    private int emphasizeAnimationDelay = 0;
+    private int emphasizeDirection = 1; // 1 for forward, -1 for reverse
+    private boolean drawTradeMarketWarning = false;
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onRenderSlot(SlotRenderEvent.Pre e) {
+        if (!(e.getScreen() instanceof ContainerScreen cs)) return;
+
+        if (cs.getTitle().getString().equals(BLACKSMITH_TITLE)) {
+            int itemIndex = cs.getMenu().getItems().indexOf(e.getSlot().getItem());
+            if (itemIndex > 24 || itemIndex < 11) return; // Not in the sell slots
```

Comment:
This requires that it have SearchableContainerProperty though, unless you're saying to add it completely separately

Reply:
I think it would be fine to have it separate as that's just some information any future feature that wants to do blacksmith things might need. Unless we introduce a BlacksmithModel then I guess it works there too